Healthcheck integration tests are flaky when workers are slow to start

## Problem

The client integration healthcheck tests (`TestHealthcheck::test_healthcheck` and `TestAsyncClient::test_healthcheck`) intermittently fail in CI because the RQ worker container isn't ready when the tests run.

Both tests assert `health.status == "healthy"` but receive `"no_workers"` when the worker hasn't registered with Redis yet:

```
AssertionError: assert 'no_workers' == 'healthy'
```

**Failing tests:**
- `packages/naas-client/tests/integration/test_client_integration.py::TestHealthcheck::test_healthcheck` (line 21)
- `packages/naas-client/tests/integration/test_client_integration.py::TestAsyncClient::test_healthcheck` (line 89)

**Observed in:** PR #435 CI run — [client-integration job](https://github.com/lykinsbd/naas/actions/runs/24527978751/job/71703321615). The MCP PR does not touch the client or healthcheck code; this is a pre-existing timing issue.

## Suggested Fix

Add a retry/backoff mechanism to the healthcheck assertions so the test waits for workers to become available rather than failing immediately on the first poll. For example:

```python
import time

def test_healthcheck(self, client: NaasClient) -> None:
    deadline = time.monotonic() + 30
    while True:
        health = client.healthcheck()
        assert isinstance(health, HealthCheckResponse)
        if health.status == \"healthy\":
            break
        if time.monotonic() > deadline:
            pytest.fail(f\"Healthcheck not healthy after 30s: status={health.status}\")
        time.sleep(2)
    assert health.components.workers.count is not None
    assert health.components.workers.count > 0
```

Alternatively, a shared `wait_for_healthy` fixture could be used by both the sync and async test classes.

## Impact

This flaky failure blocks unrelated PRs and causes the `Client` gate job to fail as a cascade.

--- a/packages/naas-client/tests/integration/test_client_integration.py
+++ b/packages/naas-client/tests/integration/test_client_integration.py
@@ -2,6 +2,7 @@
 
 from __future__ import annotations
 
+import time
 from typing import TYPE_CHECKING
 
 import pytest
@@ -13,12 +14,21 @@ from naas_client.models import HealthCheckResponse, JobStatus
 if TYPE_CHECKING:
     from naas_client.async_client import AsyncNaasClient
 
+HEALTHCHECK_TIMEOUT = 30
+HEALTHCHECK_INTERVAL = 2
+
 
 class TestHealthcheck:
     def test_healthcheck(self, client: NaasClient) -> None:
-        health = client.healthcheck()
-        assert isinstance(health, HealthCheckResponse)
-        assert health.status == "healthy"
+        deadline = time.monotonic() + HEALTHCHECK_TIMEOUT
+        while True:
+            health = client.healthcheck()
+            assert isinstance(health, HealthCheckResponse)
+            if health.status == "healthy":
+                break
+            if time.monotonic() > deadline:
+                pytest.fail(f"Healthcheck not healthy after {HEALTHCHECK_TIMEOUT}s: status={health.status}")
+            time.sleep(HEALTHCHECK_INTERVAL)
         assert health.components.workers.count is not None
         assert health.components.workers.count > 0
 
@@ -85,8 +95,16 @@ class TestSendCommand:
 class TestAsyncClient:
     @pytest.mark.asyncio
     async def test_healthcheck(self, async_client: AsyncNaasClient) -> None:
-        health = await async_client.healthcheck()
-        assert health.status == "healthy"
+        import asyncio
+
+        deadline = time.monotonic() + HEALTHCHECK_TIMEOUT
+        while True:
+            health = await async_client.healthcheck()
+            if health.status == "healthy":
+                break
+            if time.monotonic() > deadline:
+                pytest.fail(f"Healthcheck not healthy after {HEALTHCHECK_TIMEOUT}s: status={health.status}")
+            await asyncio.sleep(HEALTHCHECK_INTERVAL)
 
     @pytest.mark.asyncio
     async def test_list_contexts(self, async_client: AsyncNaasClient) -> None:

--- a/packages/naas/changes/437.testing.md
+++ b/packages/naas/changes/437.testing.md
@@ -1,0 +1,1 @@
+Add retry/backoff to healthcheck integration tests to handle worker startup timing.

